src/sentry/demo/middleware.py

```diff
@@ -31,6 +31,10 @@ def process_view(self, request, view_func, view_args, view_kwargs):
                 {"detail": "Organization creation disabled in demo mode"}, status=400
             )
 
+        # at this point, don't care about any API routes
+        if path.startswith("/api/"):
```

Comment:
this change doesn't really change spec but makes sure we don't try to log the user in for API routes which is unecessary